### ISSUE TITLE
normalize_triplet.py gets confused with non-gcc compilers

### ISSUE THREAD

**Issue by einzigsue** [JuliaLang/julia#35241]
Hi There,
I am compiling julia 1.4.0 from source and found the error message
```
$JULIAHOME/deps/tools/jldownload $JULIAHOME/deps/srccache/LibUV.v2.0.0+1.29.1-julia-0..tar.gz https://github.com/JuliaPackaging/Yggdrasil/releases/download/LibUV-v2+1.29.1-julia+0/LibUV.v2.0.0+1.29.1-julia..tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   445  100   445    0     0    489      0 --:--:-- --:--:-- --:--:--   488
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
```
where $JULIAHOME is the folder I did git checkout v1.4.0.

The url passed to jldownload is wrong, with an extra dot before .tar.gz.
Can someone quickly fix this?
Many thanks in advance
Yue

**Comment by steven-varga** [comment]
As it appears something is wrong with the download link/site. Following methods lead to the same error:

- https://github.com/JuliaLang/julia/releases/download/v1.4.0/julia-1.4.0-full.tar.gz
- https://github.com/JuliaLang/julia/releases/download/v1.3.1/julia-1.3.1-full.tar.gz
- git clone + checkout tagged master, v1.4.0, v1.3.0

Is there an alternative way to compile Julia? ie: manually downloading dependencies...

**Comment by ViralBShah** [comment]
> Is there an alternative way to compile Julia? ie: manually downloading dependencies...

`make USE_BINARYBUILDER=0`

**Comment by ViralBShah** [comment]
@einzigsue I suspect that some architecture detection is failing and that is leading to a malformed URL.

**Comment by einzigsue** [comment]
I can build 1.3.1 successfully on the same machine and this is what versioninfo() says in my 1.3.1 build.
"""
julia> versioninfo()
Julia Version 1.3.1
Commit 2d5741174c* (2019-12-30 21:36 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Xeon(R) Platinum 8268 CPU @ 2.90GHz
  WORD_SIZE: 64
  LIBM: libimf
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
"""

**Comment by einzigsue** [comment]
v1.4.1 shares the error.

**Comment by ViralBShah** [comment]
Something in `deps/tools/bb-install.mk` is not returning expected values.

**Comment by ViralBShah** [comment]
Can you try the latest master?

**Comment by shanewstone** [comment]
This is still an issue for me while trying to build Julia 1.4.2 on Arch Linux.

**Comment by shanewstone** [comment]
I should probably open a new issue, but I wanted to update my last, rather lacking, comment for posterity:

I have learned that I am experiencing the same error as above and in #35885 because I am trying to compile `julia` using `icc` and `ifort` (`USEICC=1` and `USEFCC=1`) with `USE_SYSTEM_DSFMT=0` and `USE_SYSTEM_LIBUV=0`.

If I understand correctly, then when [bb-install.mk](https://github.com/JuliaLang/julia/blob/master/deps/tools/bb-install.mk) is called from [libuv.mk](https://github.com/JuliaLang/julia/blob/master/deps/libuv.mk#L70) or [dsfmt.mk](https://github.com/JuliaLang/julia/blob/master/deps/dsfmt.mk#L70), [normalize_triplet.py](https://github.com/JuliaLang/julia/blob/master/contrib/normalize_triplet.py) apparently calls `gfortran`, which I do not have installed. This leads to a malformed URL for `libuv` and `dsfmt`.

If I install `gfortran` without changing anything else, then the error does not occur and I can compile `julia` just fine.

If I do not install `gfortran` and compile with `USE_BINARYBUILDER=0`, then I do not run into any issues and I can compile `julia` just fine.

**Comment by ViralBShah** [comment]
Should be fixed.

**Comment by EliseJ** [comment]
I ran into the same issue as shanewstone, using `icc` and `ifx`, Julia version 1.5.3.

**Comment by ViralBShah** [comment]
I'm pretty sure we don't support building julia with `icc`.

**Comment by EliseJ** [comment]
in Makefile.inc 
`# Set to 1 to use Intel C, C++, and FORTRAN compilers`
`USEICC  ?= 1`
`USEIFC  ?= 1`

**Comment by ViralBShah** [comment]
Yes, that does exist. @vtjnash @staticfloat Any thoughts if we should just remove those sections? Nobody ever builds or tests with icc/ifc regularly - so it is always a cause for concern what bugs may lurk.

**Comment by ViralBShah** [comment]
The build instructions do only mention gcc: https://github.com/JuliaLang/julia/blob/master/doc/build/build.md

**Comment by staticfloat** [comment]
I don't know anyone who has ever tried to build Julia with `icc`, so I don't know what issues there may or may not be.  Jameson may know more.

**Comment by StefanKarpinski** [comment]
There's no real benefit to supporting ICC or IFC since, aside from Julia's runtime, which isn't particularly performance sensitive, code is always generated by LLVM, regardless of which C compiler Julia was compiled with. Early on we made some efforts to support other compilers, including ICC and MSVC, but that never worked very well, has completely bitrotted, and there's no compelling reason or will to get those things working. Even if someone did get them working, we'd then have to test all those combinations in CI to keep them working. Just a whole lot of upfront and ongoing work for zero benefit. The only reasonable course of action here is to delete the stuff in make files that even suggests that ICC, IFC or any other compilers besides gcc and clang are supported.

**Comment by StefanKarpinski** [comment]
Accordingly, I've changed this to be a build system maintenance issue: someone just needs to go through the make files and delete references to ICC, IFC and any other unsupported compilers.

**Comment by ViralBShah** [comment]
I suppose MSVC should also go.

**Comment by StefanKarpinski** [comment]
Yes 👍🏻.

**Comment by ViralBShah** [comment]
@musm @vtjnash @staticfloat @vchuravy - Double checking with a few more folks here that we no longer use the MSVC stuff in the Makefiles for our Windows builds. I can put together a PR getting rid of all this stuff.

**Comment by vchuravy** [comment]
MSVC is not used or tested anywhere.

**Comment by ViralBShah** [comment]
Ok - I was just wondering if our windows build sets it or something (even though it doesn't use MSVC).
